torchdata/nodes/samplers/utils.py

```diff
@@ -10,9 +10,16 @@
 import torch.distributed as dist
 
 
-def _get_rank_seed(seed: int, generator_rank: torch.Generator, rank: int, world_size: int) -> int:
+def _get_rank_seed(seed: int, generator_rank: torch.Generator, rank: int, world_size: int, epoch: int) -> int:
     generator_rank.manual_seed(seed * world_size + rank)
-    return int(torch.randint(0, 2 ** 32 - 1, size=(1,), generator=generator_rank).item())
+    _random_seed: int = 0
+    for _ in range(epoch):
+        _random_seed = int(torch.randint(0, 2 ** 32 - 1, size=(1,), generator=generator_rank).item())
```

Comment:
```suggestion
    _random_seed = int(torch.randint(0, 2 ** 32 - 1, size=(epoch+1,), generator=generator_rank)[-1].item())
```
Is this equivalent? it would probably be faster than running torch.randint x epoch times